### PR TITLE
fix(core/group): group selected style and actions & header correction

### PR DESCRIPTION
**Pull request by goncalosard**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [ ] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [ ] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Old style:

image

Selected style is not correct according to Figma.
When item is selected, then dropdown is closed and reopened, item no longer shows clicked.
When item is selected and the dropdown is closed, it puts header selected.
Also GitHub Issue Number: #416 

## What is the new behavior?

- Added new style for selected header and item
- Item still shows as selected after dropdown is closed and reopened
- After selecting item and closing dropdown, header no longer goes as selected
- When header is selected, it no  longer moves content alignment

New style:

image

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
- [x] Click on header shoud deselect group items
- [x] Focus border was lost (@silviowolf will update to latest design spec)

**Comment by goncalosard** [comment]
@danielleroux @nuke-ellington 
Latest commit has the following changes:
- When header is selected, items are deselected.
- Focus-visible fix
- When there is no items, icon arrow is hidden. (MutationObserver added in case the user removes all items or it has no items and the user adds a item. The component needs to know if the lenght of items from the slot changed so it can update the hide or show of the icon if the slot size changed).

File: packages/core/src/components/group/group.scss [hunk 1]
```diff
@@ -25,47 +25,54 @@
...
-        border: var(--theme-primary-bdr-2);
         background-color: var(--theme-group-item--background--selected);
       }
     }
 
     @include focus-visible {
-      border-color: var(--focus--border-color);
-      border-radius: var(--theme-group--border-radius--focus);
+      border: 1px solid var(--theme-color-focus-bdr);
+      border-radius: 4px 4px 0px 0px;
```
Comment (danielleroux) [inline packages/core/src/components/group/group.scss:54]: use `--theme-default-border-radius`

`border-radius: var(--theme-default-border-radius) var(--theme-default-border-radius) 0px 0px;`

File: packages/core/src/components/group/group.scss [hunk 2]
```diff
@@ -132,6 +135,7 @@
   .btn-expand-header {
     margin: $small-space;
     margin-inline-end: $tiny-space;
+    min-width: 24px;
```
Comment (danielleroux) [inline packages/core/src/components/group/group.scss:138]: use rem

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/core/src/components/group/group.tsx [hunk 3]
```diff
@@ -184,7 +193,16 @@ export class Group {
     }
   }
 
+  private observer: MutationObserver;
+
   componentDidLoad() {
+    const slotDiv = this.hostElement.querySelector('.group-content');
+    this.observer = createMutationObserver(() => {
+      this.slotSize = this.groupItems.length;
```
Comment (danielleroux) [inline packages/core/src/components/group/group.tsx:201]: slotSize is nerver used again in the component? Do you have any local changes which are not pushed?

File: packages/core/src/components/group/group.tsx [hunk 4]
```diff
@@ -78,6 +79,8 @@ export class Group {
 
   @Element() hostElement!: HTMLIxGroupElement;
 
+  itemSelected = false;
```
Comment (danielleroux) [inline packages/core/src/components/group/group.tsx:82]: `itemSelected` is used inside `reactive typescript` part here is missing a `@State`

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/core/src/tests/group/group.e2e.ts [hunk 5]
```diff
@@ -35,7 +35,7 @@ regressionTest.describe('group', () => {
 
   regressionTest('item selected', async ({ page }) => {
     await page.goto('group/basic');
-    await page.locator('.btn-expand-header').click();
+    await page.locator('.btn-expand-header .sc-ix-icon-h').click();
```
Comment (danielleroux) [inline packages/core/src/tests/group/group.e2e.ts:38]: `.sc-ix-icon-h ` is a compiler generated class you should not use it as an selector

**Review by danielleroux — CHANGES_REQUESTED** [review]

**Review by danielleroux — APPROVED** [review]

**Review by nuke-ellington — APPROVED** [review]
lgtm

**Pull request merged**